### PR TITLE
Update HowTo guide for I2C device driver

### PR DESCRIPTION
**Pull request by btheosam**
This commit includes;
What low level implementation should provide for I2C device driver
and how the i2c device driver should be initialized and used.

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@pradeepns85 @manoharahk @samirsahu1947 Coudl you review this doc?

File: docs/HowToUsePeripheral.md [hunk 1]
```diff
@@ -242,7 +242,58 @@ static void spi_select(struct spi_dev_s *dev, enum spi_dev_e devid, bool selecte
...
+
+```
+1. EXTERN FAR struct i2c_dev_s *up_i2cinitialize(int port);
+2. EXTERN int up_i2cuninitialize(FAR struct i2c_dev_s *dev);
+3. EXTERN int up_i2creset(FAR struct i2c_dev_s *dev);
+```
+
+### I2C Initialization
+I2C can be initialzed by invoking up_i2cinitialize 
+either as part of _up_intitialize()_ or as part of _board_initialize()_.
```
Comment (manoharahk) [inline docs/HowToUsePeripheral.md]: 1. Shall we provide a link to s5j  i2c driver (arch/arm/src/s5j/s5j_i2c.c) as a reference?
2. Low level I2C driver usually bound to higher level device driver to provide a device node to application layer. Hence shall we add details about "i2c_uioregister()" which does a register driver?

File: docs/HowToUsePeripheral.md [hunk 2]
```diff
@@ -242,7 +242,58 @@ static void spi_select(struct spi_dev_s *dev, enum spi_dev_e devid, bool selecte
 
 }
 ```
+## I2C 
+
+Each board, shall;
+ 1. Expose an instanse of _struct i2c_dev_s_
```
Comment (manoharahk) [inline docs/HowToUsePeripheral.md]: Should it be instance?

File: docs/HowToUsePeripheral.md [hunk 3]
```diff
@@ -242,7 +242,58 @@ static void spi_select(struct spi_dev_s *dev, enum spi_dev_e devid, bool selecte
...
+other functions listed in **[i2c.h](../os/include/tinyara/i2c.h)**
+
+```
+1. EXTERN FAR struct i2c_dev_s *up_i2cinitialize(int port);
+2. EXTERN int up_i2cuninitialize(FAR struct i2c_dev_s *dev);
+3. EXTERN int up_i2creset(FAR struct i2c_dev_s *dev);
+```
+
+### I2C Initialization
+I2C can be initialzed by invoking up_i2cinitialize
```
Comment (manoharahk) [inline docs/HowToUsePeripheral.md]: Should it be initialized?

**Comment by pradeepns85** [comment]
@btheosam , Can we add information on how i2c clients are registered to I2C bus ?

File: docs/HowToUsePeripheral.md (hunk 1 above)
Comment (btheosam) [reply docs/HowToUsePeripheral.md]: Yes, worth mention about uio, will update this after updating the i2c code. defer this review for more than 2 weeks.

**Comment by sunghan-chang** [comment]
@pradeepns85 @btheosam I agree Pradeep's suggestion.

**Comment by btheosam** [comment]
Remember i2c is a bus (device), not a device (device as in device driver). 
I dont know if it is appropriate (for i2c) to be registereed as device driver as in current code. 
I am yet to get clarification from lsi. (About the need)

Further, as opposed to device driver's exclusive access, i2c instance dont need exclusive access, it can transact with multiple slaves at the same time (without ex access).

The only reason I see to register i2c (with a device driver) is to allow POSIX like open, read, write .. on i2c port. But, not sure if this is correct usage & needed, (coz, then user level app need to know the I2C slave address, supported freq, etc to configure.. )
If you guys can provide any need for registering i2c as device driver other than above, please enlighten me.

Further, unless **somebody clarifies** what you mean by ", Can we add information on how i2c clients are registered to I2C bus ?" I could n't comment as clients dont get registered to i2c bus. Once again, this is a bus. You just set slave addr, frequence & transact. Where is the need for registration here?
(I dont see any need for clients to register both in master & slave mode). Correct me if Iam wrong.

**Pull request closed without merging**